docs/source/docs/hmr.md

```diff
@@ -0,0 +1,121 @@
+ - tagline: How to use Hot Module Replacement with Fable-VirtualDom
+
+# Hot Module Replacement
+
+According to webpack website:
+
+> “Hot Module Replacement” (HMR) is a feature to inject updated modules into the active runtime.
+>
+> It’s like LiveReload for every module.
+
+In this documentation we are going to see how HMR can be used with Fable-VirtualDom to dynamically update the application and maintain the state between realods.
+
+
+## How to use in development mode ?
+
+First we need to install the npm dependencies and dev dependencies with: `npm install`
+
+Second we run `npm run dev` which have two effect:
+
+1. Start fable compiler in watching mode
+2. Start webpack development server for serving the files and activating the HMR
+
+Now you can access the website by navigating at [http://localhost:8080](http://localhost:8080)
+
+
+## Architecture
+
+The file `Source/static/index.js` is used has a starting point by Webpack.
+
+```js
+// Pull in desired css/stylus files
+require( '../styles/app.styl' );
+
+// Call Entry point
+var Entry = require( '../../out/Entry' );
+```
+
+The application start point is `Entry` module (js file).
+Fable is generating one *.js* file by *.fsx*.
+
+The `Entry.fsx` file is used to set up the HMR and start the VirtualDom application.
+
+```fsharp
+let contentNode = "#app"
+
+#if DEV_HMR
+
+type IModule =
+  abstract hot: obj with get, set
+
+let [<Global>] [<Emit("module")>] Module : IModule = failwith "JS only"
+
+let node = document.querySelector contentNode
+
+if isNotNull Module.hot then
+  Module.hot?accept() |> ignore
+
+  Module.hot?dispose(fun _ ->
+    node.removeChild(node.firstChild) |> ignore
+  ) |> ignore
+#endif
+
+App.Main.start contentNode ()
+```
+
+The file `Main.fsx` is a Fable-VirtualDom application with some configuration to support the HMR. 
+
+```fsharp
+type Model =
+    { Input: string }
+
+    static member initial =
+      #if DEV_HMR
+      // This section is used to maintain state between HMR
+      if isNotNull (unbox window?storage) then
+        unbox window?storage
```

Comment:
I actually think this could be solved with a subscriber, but not sure how just yet :)